### PR TITLE
start porting examples to surface syntax

### PR DESCRIPTION
**Pull request by jhellerstein**
Moved old examples to `deprecated_examples` and removed from `Cargo.toml`.

### CODE REVIEW

File: hydroflow/Cargo.toml [hunk 1]
```diff
@@ -11,24 +11,12 @@ macros = [ "hydroflow_macro", "hydroflow_datalog" ]
 [[example]]
 name = "chat"
 
-[[example]]
-name = "covid_tracing"
-
-[[example]]
-name = "covid_tracing_dist"
```
Comment (MingweiSamuel) [inline hydroflow/Cargo.toml:18]: covid_tracing and covid_tracing_dist aren't deprecated - they use the core api

File: hydroflow/examples/chat/client.rs [hunk 2]
```diff
@@ -1,112 +1,91 @@
...
+        // set up channels
+        outbound_chan = merge() -> sink_async(outbound);
+        inbound_chan = recv_stream(inbound) -> map(deserialize_msg) -> tee();
+        connect_acks = inbound_chan[0] -> filter_map(|m: Message| match m {
+            Message::ConnectResponse => Some(m),
+            _ => None }) -> tee();
+        messages = inbound_chan[1] -> filter_map(|m: Message| match m {
+            Message::ChatMsg{ nickname, message, ts } =>
+                Some(Message::ChatMsg{nickname, message, ts}),
+            _ => None });
```
Comment (MingweiSamuel) [inline hydroflow/examples/chat/client.rs:35]: Yeah can put a `//TODO` here for when we have an actual switching/demuxing operator

File: hydroflow/examples/graph_reachability/README.md [hunk 3]
```diff
@@ -1,8 +1 @@
-# Graph Reachability
-
-To run:
-```
-cargo run -p hydroflow --example graph_reachability
-```
-
-Adding the `-- --graph <graph_type>` flag to the end of the command line above will print out a node-and-edge diagram of the program. Supported values for `<graph_type>` include [mermaid](https://mermaid-js.github.io/) and [dot](https://graphviz.org/doc/info/lang.html).
```
Comment (MingweiSamuel) [inline hydroflow/examples/graph_reachability/README.md:8]: Is this intentional? seems like the deleted version should stick around. maybe accident from refactoring in the surface syntax version

**Review by MingweiSamuel — APPROVED** [review]
(Guess merge commits don't matter if you use the squash function for accepting the PR)

File: hydroflow/Cargo.toml (hunk 1 above)
Comment (jhellerstein) [reply hydroflow/Cargo.toml:18]: I'd argue that's deprecated. We're not documenting it.

File: hydroflow/examples/chat/client.rs [hunk 4]
```diff
@@ -1,112 +1,91 @@
...
+use hydroflow::hydroflow_syntax;
+use std::net::SocketAddr;
+use tokio::io::AsyncBufReadExt;
+use tokio::net::UdpSocket;
+use tokio_stream::wrappers::LinesStream;
 
 pub(crate) async fn run_client(opts: Opts) {
-    let mut hf = HydroflowBuilder::default();
+    // set up network and I/O channels
+    let server_addr: SocketAddr = format!("{}:{}", opts.addr, opts.port).parse().unwrap();
```
[the remark this answers is not in the document]
Comment (jhellerstein) [reply hydroflow/examples/chat/client.rs:16]: Actually that doesn't work. The best I could do in that style is:
```
let server_addr: SocketAddr =
        (opts.addr.parse::<std::net::Ipv4Addr>().unwrap(), opts.port).into();
```

File: hydroflow/examples/chat/client.rs (hunk 2 above)
Comment (jhellerstein) [reply hydroflow/examples/chat/client.rs]: I chose a different demuxing style for now, it's clear what's going on. The TODO is implicit!

**Pull request merged**